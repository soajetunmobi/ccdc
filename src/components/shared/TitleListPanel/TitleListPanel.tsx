import React, { FC } from 'react'

interface IDetailPanelProps {
  records: Array<RecordDetailObject>
  onSelectHandler: (id: string) => void
}

export const TitleListPanel: FC<IDetailPanelProps> = props => {
  const { records, onSelectHandler } = props
  return (
    <div className=''>
      <h2>Records</h2>
      <div className='record'>
        {records &&
          records.map((record, index) => (
            <div
              className={`record-row ${(index + 1) % 2 === 0 ? 'even' : 'odd'}`}
              key={record.id}
              onClick={() => {
                onSelectHandler(record.id)
              }}
            >
              {record.title}
            </div>
          ))}
      </div>
    </div>
  )
}
