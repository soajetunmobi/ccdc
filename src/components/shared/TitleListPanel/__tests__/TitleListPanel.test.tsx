import React from 'react'
import { TitleListPanel } from '../TitleListPanel'
import { fireEvent, render, screen } from '@testing-library/react'

describe('TitleListPanel', () => {
  const records: Array<RecordDetailObject> = [
    {
      id: 'test-id',
      title: 'Test title',
      url: 'https://test.com/test',
      subjects: [{ subject: 'Test subject' }],
      publisher: 'Test publisher',
      publicationYear: 2021
    }
  ]

  const mockedOnSelectHandler = jest.fn()

  test('it renders correctly', () => {
    const { container } = render(
      <TitleListPanel records={records} onSelectHandler={mockedOnSelectHandler} />
    )
    expect(container).toMatchSnapshot()
  })

  test('it renders title', () => {
    render(<TitleListPanel records={records} onSelectHandler={mockedOnSelectHandler} />)
    expect(screen.getByText(records[0].title)).toBeInTheDocument()
  })

  test('should be able to view detail', () => {
    render(<TitleListPanel records={records} onSelectHandler={mockedOnSelectHandler} />)
    const row = screen.getByText(records[0].title)
    expect(row).toBeInTheDocument()
    fireEvent.click(row)
    expect(mockedOnSelectHandler).toBeCalled()
  })
})
