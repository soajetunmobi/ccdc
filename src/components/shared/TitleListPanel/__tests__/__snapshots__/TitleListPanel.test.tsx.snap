// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleListPanel it renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <h2>
      Records
    </h2>
    <div
      class="record"
    >
      <div
        class="record-row odd"
      >
        Test title
      </div>
    </div>
  </div>
</div>
`;
